app/services/global_autocomplete.rb

```diff
@@ -166,4 +174,15 @@ def column_def(column)
     def scope_with_where?(scope)
       (scope[:columns].present? && scope[:columns].any?) || scope[:query_where].present?
     end
+
+    def policy_ok?
```

Comment:
here's the key change, if it's a global search, it needs both index and show, but if it's a searchable association, it just needs index